avalanche/training/supervised/der.py

```diff
@@ -0,0 +1,263 @@
+import copy
+from typing import List, Optional
+
+import numpy as np
+import torch
+import torch.nn.functional as F
+from torch.nn import CrossEntropyLoss, Module
+from torch.optim import Optimizer
+from torch.utils.data import Dataset
+from torchvision.transforms import Compose, RandomCrop, RandomHorizontalFlip
+
+from avalanche.benchmarks.utils import (classification_subset,
+                                        make_tensor_classification_dataset)
+from avalanche.core import SupervisedPlugin
+from avalanche.models.utils import avalanche_forward
+from avalanche.training.plugins.evaluation import default_evaluator
+from avalanche.training.storage_policy import (BalancedExemplarsBuffer,
+                                               ReservoirSamplingBuffer)
+from avalanche.training.templates import SupervisedTemplate
+
+
+def cycle(loader):
+    while True:
+        for batch in loader:
+            yield batch
+
+
+@torch.no_grad()
+def dataset_with_logits(dataset, model, batch_size, device, _max_size=300):
+    model = copy.deepcopy(model)
+    model.train()
+    logits = []
+    data = []
+    loader = torch.utils.data.DataLoader(
+        dataset, 
+        batch_size=batch_size, 
+        shuffle=False)
+
+    for x, _, _ in loader:
+        x = x.to(device)
+        out = model(x)
+        logits.append(out)
+        data.append(x)
+
+    logits = torch.cat(logits)
+    data = torch.cat(data)
+
+    logits = F.pad(logits, (0, _max_size - logits.shape[1]), value=0)
+
+    if len(data.shape) == 4:
+        transform = Compose(
+            [RandomCrop(data[0].shape[2], padding=4), RandomHorizontalFlip()]
```

Comment:
instead of hardcoding transforms, can we have an argument that can be set by the user?